pkg/asset/agent/image/oc.go

```diff
@@ -126,19 +138,54 @@ func (r *release) extractFileFromImage(log logrus.FieldLogger, image, file, cach
 		cmd = fmt.Sprintf(templateImageExtract, file, cacheDir, image)
 	}
 
-	log.Debugf("extracting %s to %s, %s", file, cacheDir, cmd)
+	logrus.Debugf("extracting %s to %s, %s", file, cacheDir, cmd)
 	_, err := retry.Do(r.config.MaxTries, r.config.RetryDelay, execute, log, r.executer, pullSecret, cmd)
 	if err != nil {
 		return "", err
 	}
 	// set path
 	path := filepath.Join(cacheDir, path.Base(file))
-	log.Info("Successfully extracted base ISO from the release")
-	log.Debugf("Base ISO %s cached at %s", file, path)
 	return path, nil
 }
 
-func execute(log logrus.FieldLogger, executer executer.Executer, pullSecret string, command string) (string, error) {
+// Check if there is a different base ISO in the release payload
+func (r *release) verifyCacheFile(log logrus.FieldLogger, image, file, pullSecret, architecture string, mirrorConfig []mirror.RegistriesConfig) (bool, error) {
+
+	tempDir, err := ioutil.TempDir("", "cache")
+	if err != nil {
+		return false, err
+	}
+
+	// Get the file containing the coreos sha256 hash
+	shaFilename := fmt.Sprintf(coreOsSha256FileName, architecture)
+	shaFile, err := r.extractFileFromImage(log, image, shaFilename, tempDir, pullSecret, mirrorConfig)
+	if err != nil {
+		return false, err
```

Comment:
I think this should return `false, nil`, because as it stands if we fail to extract the checksum file from the image we will fail the entire operation, but we really want to continue to fall back to trying to download the image.

Reply:
Yeah, good point, fixed.